workflow/scripts/osemosys_global/transmission/main.py

```diff
@@ -0,0 +1,247 @@
+                                                                 end_year, 
+                                                                 region_name)
+    
+    tech_set_trn_final = pd.concat([tech_set, create_tech_set_trn(interface_data)])
+
+    # Alter output csv's based on user defined capacities following user config.
+    if not tech_capacity_trn is None:
+        (tech_set_trn_final, 
+         df_max_cap_invest_trn_final, 
+         df_min_cap_invest_trn_final, 
+         df_res_cap_trn_final, 
+         df_iar_trn_final,
+         df_oar_trn_final,
+         df_op_life_trn_final, 
+         df_capact_trn_final, 
+         df_cap_cost_trn_final
+         ) = set_user_defined_capacity_trn(
+             tech_capacity_trn, 
+             op_life_dict, 
+             tech_set_trn_final, 
+             min_cap_invest_base, 
+             df_max_cap_invest_trn_final, 
+             res_cap_base,
+             df_iar_trn_final,
+             df_oar_trn_final,
+             df_op_life_trn_final,
+             df_capact_trn_final,
+             df_cap_cost_trn_final,
+             start_year,
+             end_year,
+             region_name,
+             DF_IAR_CUSTOM_VAL, 
+             DF_OAR_CUSTOM_VAL
+             )
+
+    # OUTPUT CSV's
+    
+    df_oar_trn_final.to_csv(os.path.join(output_data_dir, "OutputActivityRatio.csv"), index=None)
+    
+    df_iar_trn_final.to_csv(os.path.join(output_data_dir, "InputActivityRatio.csv"), index=None)
+    
+    df_crossborder_final.to_csv(os.path.join(output_data_dir,
+                                            "TotalTechnologyModelPeriodActivityUpperLimit.csv"),
+                                index = None)
+    
+    df_capact_trn_final.to_csv(os.path.join(output_data_dir, 
+                                            "CapacityToActivityUnit.csv"), index = None)
+    
+    df_cap_cost_trn_final.to_csv(os.path.join(output_data_dir, "CapitalCost.csv"), index = None)
+    
+    df_fix_cost_trn_final.to_csv(os.path.join(output_data_dir, "FixedCost.csv"), index = None)
+    
+    df_op_life_trn_final.to_csv(os.path.join(output_data_dir, "OperationalLife.csv"), index = None)
+    
+    df_max_cap_invest_trn_final.to_csv(os.path.join(output_data_dir, 
+                                            'TotalAnnualMaxCapacityInvestment.csv'),
+                                        index = None)
+
+    tech_set_trn_final.to_csv(os.path.join(output_data_dir, "TECHNOLOGY.csv"), index = None)
+    
+    if not tech_capacity_trn is None:
+        
+        df_res_cap_trn_final.to_csv(os.path.join(output_data_dir, 
+                                                'ResidualCapacity.csv'),
+                                            index = None)       
+        
+        df_min_cap_invest_trn_final.to_csv(os.path.join(output_data_dir, 
+                                                'TotalAnnualMinCapacityInvestment.csv'),
+                                            index = None)
```

Comment:
If these files are not printed out, Snakemake will throw an error, as it requires file specified in the rule to be printed out. Can we change this to just writing out unchanged (from the input) csv files?

Reply:
Amended